### PR TITLE
feat: redirecting to Domain signin and signup page resolved

### LINKED ISSUE #70: Create a Proxy Signin and Signup Page
As we have the original Signin and Signup page on a custom domain that is `app.localhost:3000/signin`, we require to create a `/signin` and `/signup` page on the root domain as well so that if anyone visits `localhost:3000/signin` this redirects to `app.localhost:3000/signin` and same for signup page as well.

Note: Please use the domain name from process.env to redirect the user.

### PR DESCRIPTION
**Pull request by Hari1122-code**
## What does this PR do?

Fixes #70

if anyone visits localhost:3000/signin this redirects to app.localhost:3000/signin and same for signup page as well.

https://github.com/piyushgarg-dev/review-app/assets/77155890/7017552b-68f4-4d39-a84d-5ea37f9a4dcd

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How should this be tested?

- try to go to the domain pages localhost:3000/signin or localhost:3000/signup
- it redirects to the domain page  app.localhost:3000/signin or app.localhost:3000/signup respectively

### CODE REVIEW

**Comment by Hari1122-code** [comment]
@piyushgarg-dev , could you please review this PR?

File: middleware.ts [hunk 1]
```diff
@@ -18,6 +18,10 @@ export function middleware(req: NextRequest) {
 
   if (!hostname) return
 
+  if(hostname==`${process.env.NEXT_PUBLIC_APP_DOMAIN}` && (path=="/signin" || path=="/signup")){ 
+    return NextResponse.redirect(`http://app.${process.env.NEXT_PUBLIC_APP_DOMAIN}${path}`);
+  }
```
Comment (piyushgarg-dev) [inline middleware.ts:23]: Changes Required
1. Take the `app` using process.env.NEXT_PUBLIC_APP_DASHBOARD_DOMAIN
2. Take the `https` or `http` based on the incoming request protocol.

**Review by piyushgarg-dev — CHANGES_REQUESTED** [review]

File: middleware.ts (hunk 1 above)
Comment (Hari1122-code) [reply middleware.ts]: @piyushgarg-dev, could you review this PR now, I made the changes required

Comment (piyushgarg-dev) [reply middleware.ts]: @Hari1122-code 
Prev solution was valid where you were checking /signin and /signup routes. We don't want to redirect every route to app subdomain.

Comment (Hari1122-code) [reply middleware.ts]: Yes, you are correct @piyushgarg-dev  and sorry for the extra commits now I have again resolved the changes.
Please review it

**Comment by piyushgarg-dev** [comment]
Great Work @Hari1122-code 
Would review and merge this

**Comment by Hari1122-code** [comment]
> Great Work @Hari1122-code Would review and merge this

Thank you @piyushgarg-dev , Let me know if any further changes required.

**Pull request merged**